torch/nn/modules/transformer.py

```diff
@@ -375,6 +376,145 @@ def forward(self, tgt, memory, tgt_mask=None, memory_mask=None,
         return tgt
 
 
+class MultiheadAttentionInProjection(Module):
```

Comment:
Is this a concept more generic that this? I.e. something that is used in other Attention mechanisms as well etc.

Reply:
The counterargument is that with both in/out projection modules users will have the JIT/quantization support, which may not be straightforward.
For some users who are interested to implement a custom attention layers, they don't need to re-write the in/out projection layers. For example, if they want to change the score func in the dot product layer, they can still re-use those in/out projection modules without any changes.

Reply:
I agree that this is useful, but I suspect that the name doesn't do it's wider applicability justice.